starlite/connection/request.py

```diff
@@ -136,10 +133,7 @@ async def body(self) -> bytes:
             A byte-string representing the body of the request.
         """
         if self._body is Empty:
-            chunks = []
-            async for chunk in self.stream():
-                chunks.append(chunk)
-            self._body = self.scope["_body"] = b"".join(chunks)  # type: ignore[typeddict-item]
+            self._body = self.scope["_body"] = b"".join([c async for c in self.stream()])  # type: ignore[typeddict-item]
```

Comment:
Sonar shows this as a bug on my pr branch and fails the CI - did you have to do anything to get it passed (that's why I tried to change the type to `AsyncIterator` btw.

Reply:
yes, i told sonar its wrong.